x-pack/plugins/observability/public/pages/cases/case_details.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useParams } from 'react-router-dom';
+
+import { CaseView } from '../../components/app/cases/case_view';
+import { useGetUserCasesPermissions } from '../../hooks/use_get_user_cases_permissions';
+import { useKibana } from '../../utils/kibana_react';
+import { CASES_APP_ID } from '../../components/app/cases/constants';
+import { CaseCallOut, savedObjectReadOnlyErrorMessage } from '../../components/app/cases/callout';
+
+export const CaseDetailsPage = React.memo(() => {
+  const {
+    application: { navigateToApp },
+  } = useKibana().services;
+  const userPermissions = useGetUserCasesPermissions();
+  const { detailName: caseId, subCaseId } = useParams<{
+    detailName?: string;
+    subCaseId?: string;
+  }>();
+
+  if (userPermissions != null && !userPermissions.read) {
```

Comment:
This check is for the scenario a user does not have read permission and enters through a URL not through the navigation, right? If that the case this code will redirect them to the overview page. Is that intented?

Reply:
yes i believe so, pulled over exactly from `security_solution`